sycl/test-e2e/Matrix/joint_matrix_transposeAB.cpp

```diff
@@ -0,0 +1,133 @@
+  std::cout << "SG size " << sg_size << " ";
+
+  q.submit([&](handler &cgh) {
+     cgh.parallel_for<class MT<T, TR, TC>>(
+         nd_range<2>({NR / TR, NC / TC * sg_size}, {1, 1 * sg_size}),
+         [=](nd_item<2> spmd_item)
+#ifdef SG_SZ
+             [[sycl::reqd_sub_group_size(SG_SZ)]]
+#endif
+         {
+           auto in_ptr =
+               address_space_cast<sycl::access::address_space::global_space,
+                                  sycl::access::decorated::no>(in);
+           auto out_ptr =
+               address_space_cast<sycl::access::address_space::global_space,
+                                  sycl::access::decorated::no>(out);
+
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           sub_group sg = spmd_item.get_sub_group();
+           joint_matrix<sub_group, T, Use, TR, TC, layout::row_major>
+               matrix_row_major;
+           joint_matrix<sub_group, T, Use, TR, TC, layout::col_major>
+               matrix_col_major;
+
+           auto row_major_offset =
+               (sg_startx * TR) * NC + sg_starty / sg_size * TC;
+           auto col_major_offset =
+               (sg_startx * TR) + (sg_starty / sg_size * TC) * NR;
+
+           joint_matrix_load(sg, matrix_row_major, in_ptr + row_major_offset,
+                             NC);
+           joint_matrix_copy(sg, matrix_row_major, matrix_col_major);
+           ext::intel::experimental::matrix::joint_matrix_store(
+               sg, matrix_col_major, out_ptr + col_major_offset, NR);
+         }); // parallel for
+   }).wait();
+}
+
+template <typename T, size_t TR, size_t TC, use Use> void test() {
+  std::cout << "Test " << TR << " x " << TC << " ";
+  static constexpr size_t SCALE = 2;
+  static constexpr size_t MATRIX_R = TR * SCALE;
+  static constexpr size_t MATRIX_C = TC * SCALE;
+
+  queue q;
+  T *in = malloc_shared<T>(MATRIX_R * MATRIX_C, q);
+  T *col_major = malloc_shared<T>(MATRIX_C * MATRIX_R, q);
+  T *ref_col_major = malloc_shared<T>(MATRIX_C * MATRIX_R, q);
+
+  matrix_rand(MATRIX_R, MATRIX_C, in, (T)5);
+  matrix_transpose<TR, TC, T, MATRIX_R, MATRIX_C, Use>(in, col_major, q);
+  matrix_transpose(MATRIX_R, MATRIX_C, ref_col_major, in);
+  assert((matrix_compare<T, T, true>(MATRIX_C, MATRIX_R, col_major,
+                                     ref_col_major)));
+  std::cout << "PASSED\n";
+
+  free(in, q);
+  free(col_major, q);
+  free(ref_col_major, q);
+}
+
+int main() {
+  queue q;
+  std::vector<combination> combinations =
+      q.get_device().get_info<syclex::info::device::matrix_combinations>();
+  bool bf16_run = false;
+  bool half_run = false;
+  bool int8_run = false;
+
+  for (auto &combination : combinations) {
+    if (!bf16_run && combination.atype == matrix_type::bf16) {
```

Comment:
we should add the big combinations as well.
If these are not supported yet, comment them out and add jira in the comment.

Reply:
I plan to add them later in the next PRs, when I start working on them in IGC.